lib/ecto/multi.ex

```diff
@@ -199,23 +204,33 @@ defmodule Ecto.Multi do
 
   Accepts the same arguments and options as `Ecto.Repo.update/3` does.
   """
-  @spec update(t, name, Changeset.t, Keyword.t) :: t
-  def update(multi, name, %Changeset{} = changeset, opts \\ []) do
+  @spec update(t, name, Changeset.t | changeset_fun, Keyword.t) :: t
+  def update(multi, name, changeset_or_run, opts \\ [])
```

Comment:
Let's not call the argument `changeset_or_run` - `changeset_or_fun` will be better.